enumerable.rb

```diff
@@ -0,0 +1,104 @@
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    while count < length
+      yield(self[count], count)
+      count += 1
+    end
+  end
```

Comment:
```rb
2.6.5 :020 > array
 => [5, 3, 1, 8, 5, 7, 6, 6, 0, 5, 7, 6, 8, 2, 4, 2, 6, 4, 4, 3, 0, 3, 2, 6, 8, 5, 0, 1, 5, 0, 6, 1, 6, 7, 4, 2, 4, 1, 4, 1, 6, 2, 8, 5, 2, 5, 7, 5, 7, 2, 8, 3, 2, 5, 4, 3, 8, 0, 8, 6, 3, 5, 6, 0, 3, 6, 7, 1, 4, 5, 6, 0, 7, 6, 3, 1, 3, 3, 6, 6, 3, 6, 6, 0, 6, 5, 8, 2, 8, 6, 0, 7, 4, 6, 1, 1, 2, 3, 4, 2] 
2.6.5 :021 > array.each_with_index.is_a?(Enumerator)
 => true 
2.6.5 :022 > array.my_each_with_index.is_a?(Enumerator)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):22
        1: from enumerable.rb:11:in `my_each_with_index'
LocalJumpError (no block given (yield))
2.6.5 :023 > 
```

Reply:
```rb
2.6.5 :025 > block = proc { |num| num < 4 }
 => #<Proc:0x00000000016d3a20@(irb):25> 
2.6.5 :026 > array.my_each_with_index(&block)
 => nil 
2.6.5 :027 > array.each_with_index(&block)
 => [5, 3, 1, 8, 5, 7, 6, 6, 0, 5, 7, 6, 8, 2, 4, 2, 6, 4, 4, 3, 0, 3, 2, 6, 8, 5, 0, 1, 5, 0, 6, 1, 6, 7, 4, 2, 4, 1, 4, 1, 6, 2, 8, 5, 2, 5, 7, 5, 7, 2, 8, 3, 2, 5, 4, 3, 8, 0, 8, 6, 3, 5, 6, 0, 3, 6, 7, 1, 4, 5, 6, 0, 7, 6, 3, 1, 3, 3, 6, 6, 3, 6, 6, 0, 6, 5, 8, 2, 8, 6, 0, 7, 4, 6, 1, 1, 2, 3, 4, 2] 
2.6.5 :028 >
```

Reply:
Using proc in-lieu of a block isn't part of the criteria for testing this method.
Kindly refer to the project criteria [here](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods)

Reply:
From what I read, (refers to this screenshot)
![Screenshot at 2020-07-09 09-50-50](https://user-images.githubusercontent.com/8848104/87018846-bb1c4200-c1c9-11ea-91cf-fa236a3b5dac.png)
There is nothing indicating that a bloc isn't a part of the criteria. Or am I reading wrongly ?

There is that indication
> 1. Create #my_each, a method that is identical to #each but (obviously) does not use #each. You'll need to remember the yield statement. Make sure it returns the same thing as #each as well.
> 2. Create #my_each_with_index in the same way.

That indicate us that `#each`  and `#my_each_with_index` and also others methods must return the same thing than their original method.

Is there something I am missing, please?

Reply:
Although we tried fixed all the methods to look like the actual inbuilt methods. Item 13 and 14 made it clear that a block is a requirement for item 1-12 because item 13 specifically asked for my_map to be modified to take proc, hence we could infer that item 1-12 required only a block use case, because if proc was a use case for 1-12 then item 13 because a useless requirement @maelfosso